ConcurrentModificationException in ThingManagerImpl during startup

Saw this during startup of build 3334

```
2023-02-28 08:08:16.841 [ERROR] [ore.thing.internal.ThingRegistryImpl] - Could not inform the ThingTracker 'org.openhab.core.thing.internal.ThingManagerImpl@2b2e76b' about the 'THING_ADDED' event!
java.util.ConcurrentModificationException: null
	at java.util.HashMap.computeIfAbsent(HashMap.java:1221) ~[?:?]
	at org.openhab.core.thing.internal.ThingManagerImpl.getLockForThing(ThingManagerImpl.java:948) ~[?:?]
	at org.openhab.core.thing.internal.ThingManagerImpl.registerHandler(ThingManagerImpl.java:500) ~[?:?]
	at org.openhab.core.thing.internal.ThingManagerImpl.registerAndInitializeHandler(ThingManagerImpl.java:920) ~[?:?]
	at org.openhab.core.thing.internal.ThingManagerImpl.thingAdded(ThingManagerImpl.java:357) ~[?:?]
	at org.openhab.core.thing.internal.ThingRegistryImpl.notifyTrackers(ThingRegistryImpl.java:213) ~[?:?]
	at org.openhab.core.thing.internal.ThingRegistryImpl.notifyListenersAboutAddedElement(ThingRegistryImpl.java:132) ~[?:?]
	at org.openhab.core.thing.internal.ThingRegistryImpl.notifyListenersAboutAddedElement(ThingRegistryImpl.java:1) ~[?:?]
	at org.openhab.core.common.registry.AbstractRegistry.added(AbstractRegistry.java:175) ~[?:?]
	at org.openhab.core.common.registry.AbstractRegistry.added(AbstractRegistry.java:1) ~[?:?]
	at org.openhab.core.common.registry.AbstractProvider.notifyListeners(AbstractProvider.java:60) ~[?:?]
	at org.openhab.core.common.registry.AbstractProvider.notifyListeners(AbstractProvider.java:79) ~[?:?]
	at org.openhab.core.common.registry.AbstractProvider.notifyListenersAboutAddedElement(AbstractProvider.java:83) ~[?:?]
	at org.openhab.core.model.thing.internal.GenericThingProvider.lambda$25(GenericThingProvider.java:946) ~[?:?]
	at java.util.ArrayList.forEach(ArrayList.java:1511) ~[?:?]
	at org.openhab.core.model.thing.internal.GenericThingProvider.createThingsFromModelForThingHandlerFactory(GenericThingProvider.java:949) ~[?:?]
	at org.openhab.core.model.thing.internal.GenericThingProvider.lambda$21(GenericThingProvider.java:859) ~[?:?]
	at java.util.concurrent.ConcurrentHashMap$KeySetView.forEach(ConcurrentHashMap.java:4706) ~[?:?]
	at org.openhab.core.model.thing.internal.GenericThingProvider.thingHandlerFactoryAdded(GenericThingProvider.java:861) ~[?:?]
	at org.openhab.core.model.thing.internal.GenericThingProvider.lambda$23(GenericThingProvider.java:900) ~[?:?]
	at com.google.common.collect.Iterables$4.lambda$forEach$0(Iterables.java:568) ~[?:?]
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:807) ~[?:?]
	at com.google.common.collect.Iterables$4.forEach(Iterables.java:565) ~[?:?]
	at org.openhab.core.model.thing.internal.GenericThingProvider.handleXmlThingTypesLoaded(GenericThingProvider.java:902) ~[?:?]
	at org.openhab.core.model.thing.internal.GenericThingProvider.onReadyMarkerAdded(GenericThingProvider.java:886) ~[?:?]
	at org.openhab.core.internal.service.ReadyServiceImpl.lambda$0(ReadyServiceImpl.java:52) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183) [?:?]
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) [?:?]
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179) [?:?]
	at java.util.HashMap$EntrySpliterator.forEachRemaining(HashMap.java:1850) [?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) [?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) [?:?]
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150) [?:?]
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173) [?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) [?:?]
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596) [?:?]
	at org.openhab.core.internal.service.ReadyServiceImpl.notifyTrackers(ReadyServiceImpl.java:79) [bundleFile:?]
	at org.openhab.core.internal.service.ReadyServiceImpl.markReady(ReadyServiceImpl.java:52) [bundleFile:?]
	at org.openhab.core.config.core.xml.osgi.XmlDocumentBundleTracker.registerReadyMarker(XmlDocumentBundleTracker.java:419) [bundleFile:?]
	at org.openhab.core.config.core.xml.osgi.XmlDocumentBundleTracker.finishBundle(XmlDocumentBundleTracker.java:362) [bundleFile:?]
	at org.openhab.core.config.core.xml.osgi.XmlDocumentBundleTracker.processBundle(XmlDocumentBundleTracker.java:385) [bundleFile:?]
	at org.openhab.core.config.core.xml.osgi.XmlDocumentBundleTracker$2.run(XmlDocumentBundleTracker.java:347) [bundleFile:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]
```

--- a/bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ThingManagerImpl.java
+++ b/bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ThingManagerImpl.java
@@ -14,7 +14,6 @@ package org.openhab.core.thing.internal;
 
 import java.net.URI;
 import java.text.MessageFormat;
-import java.util.HashMap;
 import java.util.HashSet;
 import java.util.Iterator;
 import java.util.List;
@@ -146,8 +145,8 @@ public class ThingManagerImpl implements ReadyTracker, ThingManager, ThingTracke
     private final Map<ThingUID, ThingPrerequisites> missingPrerequisites = new ConcurrentHashMap<>();
 
     private final Map<ThingUID, Thing> things = new ConcurrentHashMap<>();
-    private final Map<ThingUID, Lock> thingLocks = new HashMap<>();
-    private final Set<ThingUID> thingUpdatedLock = new HashSet<>();
+    private final Map<ThingUID, Lock> thingLocks = new ConcurrentHashMap<>();
+    private final Set<ThingUID> thingUpdatedLock = ConcurrentHashMap.newKeySet();
 
     protected final ChannelGroupTypeRegistry channelGroupTypeRegistry;
     protected final ChannelTypeRegistry channelTypeRegistry;

